sources/packages/web/src/components/aest/institution/modals/ApproveProgramModal.vue

```diff
@@ -0,0 +1,70 @@
+<template>
+  <ModalDialogBase
+    title="Approve program"
+    :showDialog="showDialog"
+    @dialogClosed="dialogClosed"
+  >
+    <template v-slot:content>
+      <formio
+        formName="approveEducationProgram"
+        @submitted="submitForm"
+        @loaded="formLoaded"
+      ></formio>
+    </template>
+    <template v-slot:footer>
+      <v-btn outlined :color="COLOR_BLUE" @click="dialogClosed">
+        Cancel
+      </v-btn>
+      <v-btn class="primary-btn-background" @click="approveProgram">
+        Approve now
+      </v-btn>
+    </template>
+  </ModalDialogBase>
+</template>
+
+<script lang="ts">
+import ModalDialogBase from "@/components/generic/ModalDialogBase.vue";
+import { useModalDialog } from "@/composables";
+import formio from "@/components/generic/formio.vue";
+import { COLOR_BLUE } from "@/constants";
+import { ref } from "vue";
+import { ApproveProgram } from "@/types";
+
+export default {
```

Comment:
This is the amount of code if we do not use the even and just use the Promise. It seems more clear IMO.

![image](https://user-images.githubusercontent.com/61259237/153687449-37763ba0-b39c-4fd4-bad9-14e478550947.png)

Reply:
Thanks for doing the change 😉